src/application/player.py

```diff
@@ -13,42 +13,74 @@ def __init__(self,playlist :list):
         self.song = self.playlist[0]
         self.player.set_media(self.song.media)
         self.pause = False
-        self.skip = False
+        self.next = False
         self.exit = False
+        self.skip = False
+        self.timestamp = ""
 
     def on_press(self,key):
         pressed = '{0}'.format(key)
-        if pressed == "'1'":#pause
-            if self.pause == False:
-                self.pause = True
-            else:
-                self.pause = False
-        if pressed == "'2'":#skip
-            self.skip = True
-        if pressed == "'3'":#exit1
-            self.exit = True
+        keys = ["'0'","'1'","'2'","'3'","'4'","'5'","'6'","'7'","'8'","'9'","'t'"]
+        if self.skip:
+            if pressed in keys:
+                if pressed == keys[10]:
+                    self.skip = False
+                else:
+                    self.timestamp += pressed.replace("'","")
+        else:
+            if pressed == keys[1]:#pause
+                if self.pause == False:
+                    self.pause = True
+                else:
+                    self.pause = False
+            if pressed == keys[2]:#skip
+                self.next = True
+            if pressed == keys[3]:#exit
+                self.exit = True
+            if pressed == keys[4]:#print timestamp
+                self.skip = True
 
     def start(self):
-        print("Playing: "+self.song.name+"\n1: Pause\n2: Skip\n3: Exit")
+        print("Playing: "+self.song.name+"\n1: Pause    2: Next\n3: Exit     4: Skip")
         listener = keyboard.Listener(on_press=self.on_press)
         listener.start()
         while True:
             if not self.pause:
                 self.play()
             if self.pause and self.player.is_playing():
                 self.player.set_pause(1) #TODO Send message to pause
-            if self.skip:
-                self.skip = False
+            if self.next:
+                self.next = False
                 self.next_song() #TODO Send message to skip to next song
             if self.exit:
                 break #TODO Send message that this node is leaving lobby
 
     def play(self):
         self.player.play()
         time.sleep(2) # VLC needs time to get ready
-        while (self.pause,self.skip,self.exit) == (False,False,False):
+        while (self.pause,self.next,self.exit) == (False,False,False):
+            if self.skip:
+                timestamp = self.player.get_time()/1000
```

Comment:
Could you please move this to a separate function? Because when a message arrives from the master to skip to a certain timestamp it would be easier to just call a function like self.jump_to_timestamp(timestamp) or something similar

Reply:
Done